### PR TITLE
outlier detection: event logging

### PR DESCRIPTION
**Pull request by mattklein123**

### CODE REVIEW

**Comment by mattklein123** [comment]
@lyft/network-team

File: source/common/upstream/outlier_detection_impl.cc [hunk 1]
```diff
@@ -152,5 +164,49 @@ void DetectorImpl::runCallbacks(HostPtr host) {
   }
 }
 
+void EventLoggerImpl::logEject(HostDescriptionPtr host, EjectionType type) {
+  // TODO: Log friendly host name (e.g., instance ID or DNS name).
+  // clang-format off
+  static const std::string json =
```
Comment (RomanDzhabarov) [inline source/common/upstream/outlier_detection_impl.cc]: minor, but having something like
std::string json = R"EOF(
  {
    
  }
  )EOF";
easier to read, but up to you.

Comment (mattklein123) [inline source/common/upstream/outlier_detection_impl.cc]: That includes newlines which need to be trimmed. I didn't want to bother with that.

**Comment by RomanDzhabarov** [comment]
+1

**Pull request merged**